### PR TITLE
Common: replace use of erasure with runtimeClass (Closes #252)

### LINKED ISSUE #252: Common: replace use of erasure with runtimeClass
Should fix some warnings:

``` scala
[warn] /vagrant/3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/MapTransformer.scala:94: method erasure in trait ClassManifestDeprecatedApis is deprecated: Use runtimeClass instead
[warn]     val newInst = m.erasure.newInstance()
```

### PR DESCRIPTION
**Pull request by szareiangm**
Small PR just for the little issue with runtimeClass.

### CODE REVIEW

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala [hunk 1]
```diff
@@ -438,21 +438,24 @@ object EnrichmentManager {
...
-        }
-      case None => None.asRight
-    }
+    if (List(null, "", s"\0").contains(event.useragent)) None.asRight
+    else
+      iabEnrichment match {
+        case Some(iab) =>
+          val ipAddressWithoutPort =  event.user_ipaddress match {
+            case IPv4Regex(ipv4) => ipv4
+            case ipv6 =>
```
Comment (chuwy) [inline modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala:447]: What makes us so sure it cannot be gibberish? Is there `isIPv6(s: String): Boolean` function in any of our libraries?

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala [hunk 2]
```diff
@@ -438,21 +438,24 @@ object EnrichmentManager {
...
-                Option(event.useragent),
-                Option(ipv4),
-                Option(event.derived_tstamp).map(EventEnrichments.fromTimestamp)
-              )
-              .map(_.some)
-          case _ => None.asRight
-        }
-      case None => None.asRight
-    }
+    if (List(null, "", s"\0").contains(event.useragent)) None.asRight
```
Comment (chuwy) [inline modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala:441]: Moving it out from main match is smart. 👍 

I'd maybe change it to:


```suggestion
      if (List(null, s"\0").contains(event.useragent)) || event.useragent.trim.isEmpty None.asRight
```

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala [hunk 3]
```diff
@@ -438,21 +438,24 @@ object EnrichmentManager {
...
+    else
+      iabEnrichment match {
+        case Some(iab) =>
+          val ipAddressWithoutPort =  event.user_ipaddress match {
+            case IPv4Regex(ipv4) => ipv4
+            case ipv6 =>
+              if (ipv6.contains('[') && ipv6.contains(']')) {
```
Comment (chuwy) [inline modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala:448]: Can you move that into `normalizeIPv6` function explaining why its needed. Also a test for this case would be nice.

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala [hunk 4]
```diff
@@ -438,21 +438,24 @@ object EnrichmentManager {
...
-              .map(_.some)
-          case _ => None.asRight
-        }
-      case None => None.asRight
-    }
+    if (List(null, "", s"\0").contains(event.useragent)) None.asRight
+    else
+      iabEnrichment match {
+        case Some(iab) =>
+          val ipAddressWithoutPort =  event.user_ipaddress match {
```
Comment (chuwy) [inline modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala:445]:
```suggestion
          val ipAddressWithoutPort = event.user_ipaddress match {
```

Did it pass through scalafmt?

**Comment by szareiangm** [comment]
@chuwy  apparently there was another forcepush after I created the PR. Two of the commits regarding your comments are @oguzhanunlu 's work. 

Just to make sure we are on the same page. 

I will work on es-loader changes today.

**Comment by oguzhanunlu** [comment]
Hi @szareiangm , I've cherry-pick'ed your commit into release branch and updated commit message to align with rest, instead of rebasing and merging. I needed to force push a few times on release branch, hence the confusion here.

Huge thanks for your contributions! I'll close this PR assuming no further update is needed on this.

**Pull request closed without merging**